tools/async_to_sync.py

```diff
@@ -567,6 +572,20 @@ def parse_cmdline() -> Namespace:
     parser.add_argument(
         "--all", action="store_true", help="process all the files of the project"
     )
+    parser.add_argument(
+        "-j",
+        "--jobs",
+        type=int,
+        metavar="N",
+        const=None,
+        nargs="?",
+        default=1,
+        dest="max_workers",
```

Comment:
> Some of these options seem redundant or contradictory:
> 
>     * `const` is probably not needed if the action is not `store_const`,
> 
>     * `nargs` = `?` is the default for an option
> 
>     * default = 1 seems contradictory with "pass no argument to use the number of processors on the machine". Maybe that requires to pass None to the executor?

There are all required together in order to:
- make the default behavior be single-process,
- make `-j` (no value) be "use all processors" (because const=None)
- make `-j N` work normally

See https://docs.python.org/3/library/argparse.html#const (second use case) and https://docs.python.org/3/library/argparse.html#nargs

~~~python
>>> p = parser()
>>> p.parse_args(["--jobs"])  # all processors
Namespace(check=False, all=False, max_workers=None, container=None, inputs=[])
>>> p.parse_args(["--jobs", "2"])  # 2 workers
Namespace(check=False, all=False, max_workers=2, container=None, inputs=[])
>>> p.parse_args([])  # one processor, the default
Namespace(check=False, all=False, max_workers=1, container=None, inputs=[])
~~~

Reply:
Ah, with "pass no argument" you meant a naked `-j`. Checking `man make` it's sort of consistent with that behaviour (makes declare "no limit", I assume they just have their own undocumented limiting policies).

Does creating a pool with one worker behave single-process in the sense that it doesn't fork at all, or it just creates a pool with one process?

Why don't we use max processes by default and, if any, we introduce a switch to disable multiprocessing? Is there a reason _not_ to use multiprocessing, or to use an arbitrary N?

Reply:
> * I don't see reason to override `dest`. Accessing the default `opt.jobs` seems understandable enough.

Changed that.

> Does creating a pool with one worker behave single-process in the sense that it doesn't fork at all, or it just creates a pool with one process?

From a quick read of the source code, it seems so.

> Why don't we use max processes by default and, if any, we introduce a switch to disable multiprocessing? Is there a reason _not_ to use multiprocessing, or to use an arbitrary N?

I see no reason not to use multiprocessing in general.